### ISSUE TITLE
Feature request: The second interface in instances.

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#75]
Originally reported of Google Code with ID 19.

```
It's possible to add a second interface in instances for use the same network of the secondary
interface of each node (the one used for drdb)
or in alternative:
it's possible to add a second interface in instances for use another
network (LAN private) attached to a third interfaces of each node.

cool thanks

 
```
Originally added on 2007-11-29 11:37:14 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi,

We will definitely look at this in the next major release (1.3).

iustin
```
Originally added on 2008-01-17 10:50:52 +0000 UTC.

Changed State: Accepted
Added to Milestone: Release1.3

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2008-01-21 09:35:18 +0000 UTC.

Added Labels: Type-Enhancement
Removed Labels: Type-Defect

**Comment by Ganeti-Issues-Migrator** [comment]
```
hi,

it would be great if that enhancement should be available in a minor release before
the 1.3 release.

regards,
Al.o
```
Originally added on 2008-01-21 10:44:57 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi Alessandro,

While the core code itself supports multiple NICs, none of the following does:
import/export, command line interface for adding/removing NICs, etc.

Given this, I'm not sure if implementing this in 1.2 makes sense. We will probably
first implement it in the 1.3 development branch and if it's small enough look at
backporting it.

On the other hand, if someone wants to try and write a patch for it, we would not be
against looking at it and integrating it.

(I will not mention that it is possible to modify the config file by hand to add a
new NIC. Nope, I didn't mention it.).

regards,
iustin
```
Originally added on 2008-01-21 14:24:42 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi iustin,
i've looked for a quick & dirty way... 

We taken a look into the code and modified the config.data file for the instance that
needs 2 nics. It seems everything work fine, i've done an export with gnt-backup and
in the config.ini file i've found both nics the one connected to xen-br0 and the
other to xen-br1.

[{"ip": null, "mac": "aa:00:00:ff:b2:e4", "bridge": "xen-br0"}, {"ip": null, "mac":
"aa:00:00:ff:b2:e5", "bridge": "xen-br1"}]

Do you think is dangerous? Could you immagine some critical test to do?... 
in the future I would like to develop a patch in order to support this but now we
have to setup the production environmnet with ganeti... and there is no time to
develop anithing.

kind regards,
Al.o

```
Originally added on 2008-01-22 00:16:45 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
So, it seems I was wrong about the export/import. The export code seems to handle
multiple NICs, but it does not save (as far as I see) the bridge to which a NIC is
connected. This can be fixed.

As far as it being dangerous? No, the core code is written to support multiple NICs.
Just the interface is not; a quick grep shows that the following parts of the code
assume only one interface: gnt-instance add, gnt-instance modify, gnt-instance list,
gnt-instance info. So if you need to modify the bridge, you can't using the command
line, you would have (as you did) to modify the config file.

I think that if your instance(s) survives multiple starts/stops and failovers (in
case of mirrored disk layouts) then you are fine. Be sure to test import/export too
and see if the assignement of NICs to bridges is well (I expect this will fail).

And let us know how it goes.

thanks,
iustin
```
Originally added on 2008-01-22 00:49:35 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi iustin,
excuse me for my late reply.

My test environment (that by the end of the month sould be my production env) :
- 5 node cluster
- DRBD 0.7
- Ganeti 1.2.1
- a mix of plain and remote_raid1 instance
- all instance with 2 interface inserted by hand in configutation file

I've done all the possible test and it seems everything goes right. As you said the
export process will exports all the info about multiple nics except the bridge, the
import process simply ignore that info and bring up the instance with a single
interface. The second needs be added by hand.

I think, we'll try to take a look to the code and see how much is hard to make a patch.

thanks,
al.o

```
Originally added on 2008-01-28 09:47:13 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2008-07-03 18:21:53 +0000 UTC.

Added to Milestone: Release2.0
Removed from Milestone: Release1.3

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi!

I've read this comments, and I'm very interested in having a second network interface
in my ganeti instances too. I can't find the config.data file you spoke about to
modify the config. Could anyone tell me where can I find the file/s that i have to
modify?

Has anyone find any risk of having a second interface set up this way?

I'm using ganeti 1.2.5
Thanks. 
```
Originally added on 2008-08-07 10:32:29 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Usually in /var/lib/ganeti, also check /usr/local/var/ganeti.
```
Originally added on 2008-08-08 14:11:10 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi!

I'd like to report 'great success' using two interfaces; we have been running with
this setup for some time now, works like a charm. 
We are using Ganeti 1.2.5. Having this set up is pretty simple; adding another set of
lines specifying another NIC (in our case using a different bridge) does the trick.
As of now the export tool seems to keep information about the second NIC, but this
won't be picked up on import. 
```
Originally added on 2008-09-15 08:27:21 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
This was implemented and should work correctly in release 2.0.

Marking as fixed.
```
Originally added on 2009-08-05 08:52:29 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-10-22 08:57:07 +0000 UTC.

Changed State: Released
